dsp-role/core/src/main/scala/dsp/role/api/RoleRepo.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright © 2021 - 2022 Data and Service Center for the Humanities and/or DaSCH Service Platform contributors.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package dsp.role.api
+
+import dsp.role.domain.Role
+import dsp.valueobjects.Id._
+import zio._
+import zio.macros.accessible
+
+/**
+ * The trait (interface) for the role repository.
+ * The role repository is responsible for storing and retrieving roles.
+ * Needs to be used by the role repository implementations.
+ */
+@accessible
+trait RoleRepo {
+
+  /**
+   * Writes a role into the repository, while both creating or updating a role.
+   *
+   * @param r the role to write
+   * @return Unit
+   */
+  def storeRole(r: Role): UIO[RoleId]
```

Comment:
In the case then the parameter and its type are the same and it's strongly typed, it could be named: r, role, value, whatever and one will be always notified that's the `Role` type needs to be passed into the method.

Reply:
In strongly typed languages the type of param, if not omitted, speaks for itself. Therefore  in most of the cases instead of repeating the type in the name it could be shortened to first letter of the type, or named: value, param, whatever, because at the end the IDE will inform one need to pass something of the defined (`Role` in this case) type.